src/fiber/channel.rs

```diff
@@ -4088,182 +4071,231 @@ impl ChannelActorState {
+        let to_local_output_data = settlement_tx.outputs_data().get(0).unwrap();
+        let to_remote_output = settlement_tx.outputs().get(1).unwrap();
+        let to_remote_output_data = settlement_tx.outputs_data().get(1).unwrap();
+        let args = commitment_tx
+            .outputs()
+            .get(0)
+            .unwrap()
+            .lock()
+            .args()
+            .raw_data();
+        let message = blake2b_256(
+            [
+                to_local_output.as_slice(),
+                to_local_output_data.as_slice(),
+                to_remote_output.as_slice(),
+                to_remote_output_data.as_slice(),
+                &args[0..36],
+            ]
+            .concat(),
         );
-        verify_ctx.verify(signature, msg.as_slice())?;
+        verify_ctx.verify(commitment_tx_partial_signature, message.as_slice())?;
+
         Ok(PartiallySignedCommitmentTransaction {
-            msg,
             version: self.get_current_commitment_number(false),
-            tx,
-            signature,
-            witnesses,
+            commitment_tx,
+            funding_tx_partial_signature,
+            commitment_tx_partial_signature,
         })
     }
 
     pub fn build_and_sign_commitment_tx(
         &self,
     ) -> Result<PartiallySignedCommitmentTransaction, ProcessingChannelError> {
-        let sign_ctx = Musig2SignContext::from(self);
-
-        let (tx, msg, witnesses) = self.build_commitment_tx(true);
+        let (commitment_tx, settlement_tx) = self.build_commitment_and_settlement_tx(true);
 
-        debug!(
-            "Signing commitment tx with message {:?}",
-            hex::encode(msg.as_slice())
-        );
-        let signature = sign_ctx.sign(msg.as_slice())?;
-        debug!(
-            "Signed commitment tx ({:?}) message {:?} with signature {:?}",
-            &tx,
-            hex::encode(msg),
-            &signature,
+        let sign_ctx = Musig2SignContext::from(self);
+        let funding_tx_partial_signature = sign_ctx.sign(commitment_tx.hash().as_slice())?;
+
+        let sign_ctx = Musig2SignContext::from((self, true));
+        let to_local_output = settlement_tx.outputs().get(0).unwrap();
+        let to_local_output_data = settlement_tx.outputs_data().get(0).unwrap();
+        let to_remote_output = settlement_tx.outputs().get(1).unwrap();
+        let to_remote_output_data = settlement_tx.outputs_data().get(1).unwrap();
+        let args = commitment_tx
+            .outputs()
+            .get(0)
+            .unwrap()
+            .lock()
+            .args()
+            .raw_data();
+        let message = blake2b_256(
+            [
+                to_local_output.as_slice(),
+                to_local_output_data.as_slice(),
+                to_remote_output.as_slice(),
+                to_remote_output_data.as_slice(),
+                &args[0..36],
+            ]
+            .concat(),
         );
```

Comment:
this code logic seems duplicated with the code in `build_and_verify_commitment_tx`, a helper function `message_from_settlement_tx(..)` may be help.